circle.yml

```diff
@@ -18,7 +18,7 @@ jobs:
           name: Build docker image and push to repo
           command: |
             docker version
-            docker build -t app:build -f Dockerfile.deploy .
+            docker build -t app:build -f Dockerfile.py3.deploy .
```

Comment:
So, the main problem with that approach (with the other change below in the same file) is that we'd build `latest` (for dev) *exclusively* for python 3, and tags (for stage/prod) *exclusively* for python 2.

Also note that currently, for local dev, we use a different dockerhub repo (https://hub.docker.com/r/addons/addons-server instead of https://hub.docker.com/r/mozilla/addons-server) that is configured to build `latest` and `latest-py3` (it's not currently configured to build a Python 3 image for tags/branches, but it could be trivially by just adding it to the docker hub settings). It'd be nice to be consistent (it'd be even nicer to use the same repo for everything, but eh, baby steps).

So, I think we need to do the following:
- Build `latest` (still py2) and `latest-py3` (obviously py3) with circle ci. Same for tags.
- Deploy `latest-py3` to dev (`latest` would be unused except for if/when we need to switch back) and don't change deploy for the rest for now.

Reply:
Sounds good.  What lines are you suggesting be changed exactly?

Reply:
I'm not sure :) @autrilla do you know how we can do what I propose above efficiently? Can this be done in parallel ? Wouldn't want to make the deploys slower.

Reply:
I reversed the Dockerfile changes but I'm not clear how it affects the way forward here.  We _do_ want the docker image deployed on dev to be the py3 one.

Reply:
> I'm not sure :) @autrilla do you know how we can do what I propose above efficiently? Can this be done in parallel ? Wouldn't want to make the deploys slower.

This can certainly be done in parallel with the (relatively) new CircleCI workflows. E.g. autograph runs several things in parallel:

- [CircleCI config](https://github.com/mozilla-services/autograph/blob/master/.circleci/config.yml)
- [Example workflow](https://circleci.com/workflow-run/3f354e0b-275c-433b-9731-382cd2e95a13), shows a bunch of things running in parallel

Reply:
We're already using workflows so it doesn't sound like we'd need to add much. We only need to care about the master branch deploys right now, so I think we need to:
- Split `build-deploy` into 2 parts: a build part and a deploy part
- Clone the build part, do a py2 and a py3 version
- Make the deploy part require both builds in the workflow, and make it push both `${DOCKERHUB_REPO}:latest` and `${DOCKERHUB_REPO}:latest-py3`.

Bonus point for folding the build-release into the same jobs (just handling the fact that the docker tag is different) since it's essentially copy/pasted right now.

Once this is done, have ops make our pipeline on their end use the `latest-py3` image - but crucially, we'll have both py2 and py3 versions available, letting us switch to either if needed.